services/shhext/service.go

```diff
@@ -147,7 +148,13 @@ func (s *Service) InitProtocol(address string, password string) error {
 		os.Remove(v3Path)
 	}
 
-	persistence, err := chat.NewSQLLitePersistence(v3Path, hashedPassword)
+	// Fix IOS not encrypting database
+	if err := chat.MigrateDBFile(v3Path, v4Path, "", hashedPassword); err != nil {
+		os.Remove(v3Path)
```

Comment:
as i understand db will be removed if migration is not successful and it seems it is not going to be successful on Android (because passphrase actually worked there), do we handle this case?

Reply:
I believe key/rekey should not throw exceptions if the key is the wrong one (only querying an encrypted database will do), but will double check that to make sure it upgrades cleanly on android.

Reply:
confirmed, migrates ok. The app though crashes on logout consistently, not sure if it's related to any of these changes (I would not think so), but I'll investigate.

Reply:
it's not related to this chang for sure, it crashes in my PR too. Better to check if it crashes before my commit

Reply:
it's probably some changes in status-go that we pulled in, best guess

Reply:
probably this actually, I can see the stacktrace https://github.com/status-im/status-go/commit/54022561f5ff75de15591968b56be910f4d323b6

Reply:
@dshulyak we are getting an error on logout:

```
-10 14:50:08.296  6389     0 E Go      : 
01-10 14:50:08.296  6389     0 E Go      : goroutine 2662 [running]:
01-10 14:50:08.296  6389     0 E Go      : github.com/status-im/status-go/discovery.(*Rendezvous).Stop(0xb23c7e60, 0x0, 0x0)
01-10 14:50:08.296  6389     0 E Go      : 	/ext-go/1/src/github.com/status-im/status-go/discovery/rendezvous.go:96 +0x6e
01-10 14:50:08.296  6389     0 E Go      : github.com/status-im/status-go/discovery.Multiplexer.Stop(0xb236e830, 0x2, 0x2, 0x0, 0x0)
01-10 14:50:08.296  6389     0 E Go      : 	/ext-go/1/src/github.com/status-im/status-go/discovery/muxer.go:51 +0x62
01-10 14:50:08.296  6389     0 E Go      : github.com/status-im/status-go/node.(*StatusNode).stopDiscovery(0xb23364b0, 0xb567b774, 0xd2581f81)
01-10 14:50:08.296  6389     0 E Go      : 	/ext-go/1/src/github.com/status-im/status-go/node/status_node.go:375 +0x3f
01-10 14:50:08.296  6389     0 E Go      : github.com/status-im/status-go/node.(*StatusNode).stop(0xb23364b0, 0xd378ab01, 0xb23364b0)
01-10 14:50:08.296  6389     0 E Go      : 	/ext-go/1/src/github.com/status-im/status-go/node/status_node.go:339 +0x1a5
01-10 14:50:08.296  6389     0 E Go      : github.com/status-im/status-go/node.(*StatusNode).Stop(0xb23364b0, 0x0, 0x0)
01-10 14:50:08.296  6389     0 E Go      : 	/ext-go/1/src/github.com/status-im/status-go/node/status_node.go:333 +0x79
01-10 14:50:08.296  6389     0 E Go      : github.com/status-im/status-go/api.(*StatusBackend).stopNode(0xb24a2e40, 0x0, 0x0)
01-10 14:50:08.296  6389     0 E Go      : 	/ext-go/1/src/github.com/status-im/status-go/api/backend.go:196 +0x66
01-10 14:50:08.296  6389     0 E Go      : github.com/status-im/status-go/api.(*StatusBackend).StopNode(0xb24a2e40, 0x0, 0x0)
01-10 14:50:08.296  6389     0 E Go      : 	/ext-go/1/src/github.com/status-im/status-go/api/backend.go:188 +0x60
01-10 14:50:08.296  6389     0 E Go      : github.com/status-im/status-go/api.(*StatusBackend).StopNode-fm(0x1, 0x0)
01-10 14:50:08.296  6389     0 E Go      : 	/ext-go/1/src/github.com/status-im/status-go/lib/library.go:47 +0x1a
01-10 14:50:08.296  6389     0 E Go      : github.com/status-im/status-go/api.RunAsync.func1(0xb5bb9a40, 0xb475db00)
01-10 14:50:08.296  6389     0 E Go      : 	/ext-go/1/src/github.com/status-im/status-go/api/utils.go:7 +0x17
01-10 14:50:08.297  6389     0 E Go      : created by github.com/status-im/status-go/api.RunAsync
```

probably some nil reference:

```
01-10 14:50:08.297  6389  6429 D StatusNativeLogs: panic: runtime error: invalid memory address or nil pointer dereference
01-10 14:50:08.297  6389  6429 D StatusNativeLogs: [signal SIGSEGV: segmentation violation code=0x1 addr=0
```

Checking the code at first sight seems all ok, any idea?

Reply:
actually it might have been fixed already in develop, pushing out a different build

Reply:
@rasom @dshulyak it is fixed in develop, app does not crash with rebased version.